code/ac/llama/Model.cpp

```diff
@@ -33,13 +33,19 @@ llama_model_params llamaFromModelParams(const Model::Params& params, ModelLoadPr
 }
 } // namespace
 
-Model::Model(const char* pathToGguf, ModelLoadProgressCb loadProgressCb, Params params)
+
+Model::Model(const char* pathToGguf, std::span<std::string> loras, ModelLoadProgressCb loadProgressCb, Params params)
     : m_params(astl::move(params))
-    , m_lmodel(llama_load_model_from_file(pathToGguf, llamaFromModelParams(m_params, loadProgressCb)), llama_free_model)
 {
+    m_lmodel = ModelRegistry::getInstance().loadModel(pathToGguf, std::move(loadProgressCb), m_params);
```

Comment:
This deduplication shouldn't happen here. We want the library to be low level and allow manual lora adapter management. `Model` is a wrapper of `llama_model` and just like `llama_model` has notion of lora adapters `Model` shouldn't as well

Reply:
Expose the ModelRegistry to the plugin and changed the Model's c-tor to obtain prepared shared_ptr of a llama_model